ui/react-components/components/EditAppointment/EditAppointment.jsx

```diff
@@ -199,18 +258,43 @@ const EditAppointment = props => {
         serviceError: !appointmentDetails.service,
         startTimeError: !appointmentDetails.startTime,
         endTimeError: !appointmentDetails.endTime,
-        startTimeBeforeEndTimeError: !startTimeBeforeEndTime
+        startTimeBeforeEndTimeError: !startTimeBeforeEndTime,
+        appointmentDateError: !appointmentDetails.appointmentDate
     });
 
-    const generateAppointmentDetails = async () => {
+    const isValidRecurringAppointment = () => {
+        const startTimeBeforeEndTime = isStartTimeBeforeEndTime(appointmentDetails.startTime, appointmentDetails.endTime);
+
+        updateCommonErrorIndicators(startTimeBeforeEndTime);
+        updateErrorIndicators({
+            recurrencePeriodError: !appointmentDetails.period || appointmentDetails.period < 1
+
+    });
+        if (appointmentDetails.endDateType === 'On') {
+            updateErrorIndicators({
+                endDateError: !appointmentDetails.recurringEndDate
+            })
+        } else {
+            updateErrorIndicators({
+                occurrencesError: !appointmentDetails.occurrences || appointmentDetails.occurrences < 1
+            })
+        }
+        return appointmentDetails.service && appointmentDetails.startTime && appointmentDetails.endTime && startTimeBeforeEndTime
+            && appointmentDetails.appointmentDate && isValidEndDate();
+    };
```

Comment:
This function is checking for a valid recurring appointment and also updating error messages, better separating functionality.